src/dali/plugin/pair_converter/ccxt.py

```diff
@@ -196,73 +216,90 @@ def _bfs_cyclic(self, graph: Dict[str, Dict[str, None]], start: str, end: str) -
         # No path found
         return None
 
-    def get_historic_bar_from_native_source(self, timestamp: datetime, from_asset: str, to_asset: str, exchange: str) -> Optional[HistoricalBar]:
```

Comment:
From here on the diff becomes a bit hard to read, probably because some code got moved around, causing the diff to become more complex. Is there any chance you could restore the original order of the code and then do another PR with code reorg later?

Reply:
I needed to bundle this up into a function in order to test it. Maybe if I move it to the end?

Reply:
I think if you keep the same function order as in the original file, the diff might become easier to read. Let's try that: if it doesn't work I'll manage.

Reply:
I think what happened is the new function `_add_exchange_to_memcache` looks too similar to `get_historic_bar_from_native_source` and they are in the same place. I moved the function toward the end. I'm not sure if that helps or if I just made a bigger mess of it.

I did start using Sublime Merge to help me see if the diff is too out of wack before I commit. I'll try to keep an eye on it.